packages/block-library/src/post-featured-image/style.scss

```diff
@@ -1,4 +1,4 @@
-.wp-block-post-featured-image {
+:where(.wp-block-post-featured-image) {
```

Comment:
Thanks for working on this @dsas.

I'd appreciate thoughts from others (@ellatrix, @aaronrobertshaw, @andrewserong, @ramonjd) around the change.

I think my concern would be that it lowers the specificity across the board by 0,1,0, but the inconsistency in the specificity still remains across lots of the rules.

The `&:where(.alignleft, .alignright)` rules at the bottom are an example, they now become zero specificity, which could be risky and might cause some issues with other themes. On the other hand, the specificity of some other styles is still quite high so might not be solving the issue.

I think there are two options:
- reduce the scope of the change to just apply to the rule that's causing problems for you
- or refactor the file a bit more widely to have more consistency in specificity

In both cases, a 0,1,0 specificity for block styles is the recent target.

Reply:
Which styles exactly are being out-specified by theme.json styles?

In a very basic empty theme I added some styles to the featured image block:

```json
	"styles": {
		"blocks": {
			"core/post-featured-image": {
				"spacing": {
					"margin": {
						"top": "1.1em",
						"bottom": "1.2em",
						"left": "1.3em",
						"right": "1.4em"
					}
				}
			}
		}
	}
```

Here's what I'm seeing:

| .wp-block-post-featured-image | :where(.wp-block-post-featured-image) |
|--------|--------|
| <img width="380" alt="before-post-featured-image" src="https://github.com/user-attachments/assets/808a6fd2-3703-40b9-8b34-e25a57a25781"> | <img width="380" alt="after-post-featured-image" src="https://github.com/user-attachments/assets/60daab13-fe3b-4b7a-b170-d7c8012ee1d1"> |

The reduced specificity does give common.css internal `:where(figure)` styles a little bump upwards.

Reply:
Good discussion!

> The &:where(.alignleft, .alignright) rules at the bottom are an example, they now become zero specificity,

Good catch, maybe they wouldn't need the `:where()` wrapper anymore with this change 🤔 

> On the other hand, the specificity of some other styles is still quite high so might not be solving the issue.

In the case of the higher specificity ones, is that mostly the overlay dimming rules? I imagine those are meant to be pretty strict, so might be fine to be a little higher?

> reduce the scope of the change to just apply to the rule that's causing problems for you
> or refactor the file a bit more widely to have more consistency in specificity

These could potentially be complementary if we move the rule that needs adjusting to a separate `:where(.wp-block-post-featured-image) {` selector rather than updating this whole block? Then we might be able to do the refactoring iteratively if it feels risky to do everything at once?

Reply:
Happy to take a more cautious approach 👍 

> Which styles exactly are being out-specified by theme.json styles?

I tested by using the CSS key, so for this example lifted from the luminance theme

```json
"styles": {
		"blocks": {
			"core/post-featured-image": {
				"css": ".wp-block-post-featured-image img{vertical-align:top;width: auto;}"
			}
		}
	}
```

Reply:
> reduce the scope of the change to just apply to the rule that's causing problems for you

I think this might be the best option.

This PR's current approach will reduce all this block's root styles to zero specificity which might mean other styles could bleed in here from things like reset or other external stylesheets.

The specificity of custom CSS for blocks is capped at `0-1-0` so nested style variations can apply the correct styles based on style load order. 

The catch, as illustrated here, is that specificity limits the effectiveness of custom CSS to override overly specific block library styles. Any block library style higher than `0-1-0` specificity could be a target of global styles custom CSS. Some block library styles might need to be enforced while others won't; such as this block's inner img vertical alignment.

Eventually, we'll be able to re-work the style system based on CSS layers once all styles in WordPress can be forced into a CSS layer (a proposal for this in the CSS spec is still under consideration). Until then, some spot fixes might be required to block library styles or workarounds applied in themes.

One possible workaround here would be to move the custom CSS definition to the global custom CSS property within theme.json. In that location, the selectors specified do not have their specificity capped. 

1. Clear any custom CSS in the post featured image block type global styles
2. Navigate to Styles, then Additional CSS from the sidebar's vertical ellipsis menu
3. Add the previous custom CSS here

This resulted in the following for me:


<img width="1512" alt="Screenshot 2024-08-14 at 4 38 31 PM" src="https://github.com/user-attachments/assets/4413ca04-76f6-4949-ae7b-d871243f45c3">

Reply:
I've pushed a commit with the minimal changes I need right now. Dimming and duotone still work fine for me.

I'm ok with working around it in the theme for now too though if that's preferred, that will solve the problem for **me**.

Reply:
Reducing the specificity to `0-0-1` might still be going too far. I've pushed a small tweak to align the specificity here more with Global Styles, capping it at `0-1-0`.